MDANSE_GUI/Src/MDANSE_GUI/Tabs/Visualisers/PlotHolder.py

```diff
@@ -124,17 +128,35 @@ def plotter(self):
             LOG.error(f"PlotWidget is missing for tab {tab_id}")
             LOG.error(self._plotter)
 
-    @Slot(object)
-    def update_plot_details(self, input):
-        """This will change the line colour, thickness, etc.
-        At the moment it doesn't do anything."""
-
     @Slot()
     def update_plots(self):
-        """This will change the line colour, thickness, etc.
-        At the moment it doesn't do anything."""
+        """Update all plots.
+
+        This is typically needed if the global matplotlib settings have been changed."""
         for plotter in self._plotter:
             try:
                 plotter.plot_data(update_only=True)
-            except Exception as e:
-                LOG.error(f"Plotting failed: {e}")
+            except Exception:
+                LOG.error("Plotting failed: %s", traceback.format_exc())
+                plotter.plot_blank()
+
+    @Slot("quint64")
+    def update_plot(self, plot_number: int):
+        """Update the plot in a specific PlotWidget.
+
+        Updates the plot only in the PlotWidget matching the object ID of the
+        PlottingContext which requested the update.
+
+        Parameters
+        ----------
+        plot_number : int
+            Object id of the PlottingContext in which the settings got changed.
+        """
+        for plotter in self._plotter:
+            if plotter.unique_id == plot_number:
+                try:
+                    plotter.plot_data(update_only=True)
+                except Exception:
+                    LOG.error("Plotting failed: %s", traceback.format_exc())
+                    plotter.plot_blank()
+                    return
```

Comment:
I still think `self._plotter` may want to be a `dict[id, plotter]` so you can always iterate over `values()` where you need all of them.
```suggestion
        plotter = more_itertools.first_true(self._plotter, pred=lambda plot: plot.unique_id == plot_number)
        if plotter is None:
            return
        try:
            plotter.plot_data(update_only=True)
        except Exception:
            LOG.error("Plotting failed: %s", traceback.format_exc())
            plotter.plot_blank()
```
Is this clearer? It's certainly flatter.

Reply:
Changing `self._plotter` to a dictionary would require us to change the way we associate individual plot tabs with the `Plotter` instances. `self._plotter` being a `list` has its advantages when we start deleting plot tabs. `QTabWidget` does not assign permanent keys or id numbers to individual plot tabs, so deleting a tab in the middle re-numbers all the following ones. This is consistent with the indexing of `list` elements.

I am sure that the changes you suggested are possible, but I would prefer this PR to focus just on making sure that plotting works.